warplib/src/conversions/integer_conversions.cairo

```diff
@@ -36,3 +39,34 @@ fn bool_into_felt252(val: bool) -> felt252 {
         0
     }
 }
+
+// Cairo-1.0 corelib code from the RC
+impl U256TryIntoFelt252 of TryInto<u256, felt252> {
+    fn try_into(self: u256) -> Option<felt252> {
+        let FELT252_PRIME_HIGH = 0x8000000000000110000000000000000_u128;
+        if self.high > FELT252_PRIME_HIGH {
+            return Option::None(());
+        }
+        if self.high == FELT252_PRIME_HIGH {
+            // since FELT252_PRIME_LOW is 1.
+            if self.low != 0 {
+                return Option::None(());
+            }
+        }
+        Option::Some(
+            self.high.into() * 0x100000000000000000000000000000000_felt252 + self.low.into()
+        )
+    }
+}
+
+fn unsafe_contract_address_from_u256(x: u256) -> ContractAddress {
+    match x.try_into() {
+        Option::Some(felt_x) =>
+            match starknet::contract_address_try_from_felt252(felt_x) {
+                Option::Some(address) => address,
+                Option::None(_) =>
+                    panic_with_felt252('the felt cannot be an address'),
+            },
+        Option::None(_) => panic_with_felt252('the felt cannot be an address'),
```

Comment:
The error message is misleading, the problem is the conversion `u256 -> felt` here, not `felt -> address`

Reply:
Absolutely right! Fixed this